### DISCUSSION TITLE
Why isn't SVM and Naive Bayes included in AutoGluon-Tabular?

### DISCUSSION THREAD

**Discussion by 2catycm** [autogluon/autogluon#3489] [Q&A] [answered]
When I learn the Machine Learning course, I learnt a concept called "10 important traditional machine learning algorithm", of which supervised tabular methods includes KNN, Decision Tree, SVM, Naive Bayes, NN or Logistic Regression. 
In AutoGluon, it seems tree based methods are very prefered, both random forest, XGBoost, and other famous GBDTs are included. Also, KNN is included. Neural Network is famous in DL, so it is included too. But I did not see SVM and  Naive Bayes based methods,
Therefore, I am curious that is there an internal experimental or theoratical guideline , practice, or paper, that illustrats it is not good to include such methods into the stacking ensemble pipeline of AutoGluon-Tabular, while using tree based method are excellent?

**Comment by mglowacki100** [comment, marked as answer]
I'm not an autogluon author but SVM doesn't scale well with number of datapoint (btw. intelex version is better in this regard in comparison to plain sklearn). Regardin NB from my subjective experience NB very often add nothing to ensemble or poison it. 
Anyway, here are two tutorials how to add custom model https://auto.gluon.ai/stable/tutorials/tabular/advanced/index.html and you can experiment with it.

**Reply by apavlo89** [reply]
Svm is great for low sample data and can outperform a lot of other ones that perform well with loads of data. In my experience adding it to ensemble voting has improved accuracy in my 2k samples dataset. Not everyone is using ml with millions of samples
